### PR TITLE
feat: allow user to change profile image

### LINKED ISSUE #430: allow user to select and update profile picture
working on it :+1

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #430

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 1]
```diff
@@ -43,6 +70,45 @@ class EditProfileFragment : Fragment() {
         return rootView
     }
 
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: the variable name is confusing as you are calling  `data.data` later

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 2]
```diff
@@ -43,6 +70,45 @@ class EditProfileFragment : Fragment() {
...
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == PICK_IMAGE_REQUEST && resultCode == Activity.RESULT_OK && data != null && data.data != null) {
+            val imageUri = data.data
+            Timber.d(imageUri!!.toString())
```
[the remark this answers is not in the document]
Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: won't `"${imageUri.toString()}"` work ?

**Comment by simarsingh24** [comment]
@nikit19 resolved

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 3]
```diff
@@ -43,6 +69,44 @@ class EditProfileFragment : Fragment() {
...
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, intentData: Intent?) {
+        super.onActivityResult(requestCode, resultCode, intentData)
+        if (requestCode == PICK_IMAGE_REQUEST && resultCode == Activity.RESULT_OK && intentData != null && intentData.data != null) {
+            val imageUri = intentData.data
+            var imageStream: InputStream? = null
+            try {
+                imageStream = activity?.contentResolver?.openInputStream(imageUri)
+            } catch (e: FileNotFoundException) {
+                e.printStackTrace()
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: use timber

File: app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt [hunk 4]
```diff
@@ -43,6 +70,44 @@ class EditProfileFragment : Fragment() {
         return rootView
     }
 
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, intentData: Intent?) {
+        super.onActivityResult(requestCode, resultCode, intentData)
+        if (requestCode == PICK_IMAGE_REQUEST && resultCode == Activity.RESULT_OK && intentData != null && intentData.data != null) {
```
[the remark this answers is not in the document]
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/auth/EditProfileFragment.kt]: `requestCode == PICK_IMAGE_REQUEST && resultCode == Activity.RESULT_OK && intentData?.data != null`

**Comment by simarsingh24** [comment]
@iamareebjamal @nikit19 @dreadpool2 please review

**Pull request merged**